deployment.go

```diff
@@ -70,7 +69,7 @@ var (
 )
 
 type deploymentStore interface {
-	List() (deployments []v1beta1.Deployment, err error)
+	List() []interface{}
 }
```

Comment:
`List() []interface{}` is super generic now. Not really necessary to keep one type per resource type anymore.
Maybe just:

```go
type lister interface {
    List() []interface{}
}
```